flocker/node/agents/ebs.py

```diff
@@ -609,7 +609,10 @@ def compute_instance_id(self):
         """
         Look up the EC2 instance ID for this node.
         """
-        return get_instance_metadata()['instance-id'].decode("ascii")
+        instance_id = get_instance_metadata().get('instance-id', None)
+        if not instance_id:
```

Comment:
Last job had v. strong style constraints against `foo is None` (preferring `not foo`) unless there was a particular reason for it. Am happy to change to `is None` here, but don't see the merit of one over the other. 

Likewise, I don't see much meaningful distinction between `get`-with-out-of-band-default and a `KeyError` handler.